tests/components/balboa/test_climate.py

```diff
@@ -0,0 +1,272 @@
+"""Tests of the climate entity of the balboa integration."""
+
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.balboa.const import DOMAIN as BALBOA_DOMAIN, SIGNAL_UPDATE
+from homeassistant.components.climate.const import (
+    ATTR_FAN_MODE,
+    ATTR_HVAC_ACTION,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_PRESET_MODES,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.setup import async_setup_component
+
+from . import init_integration_mocked
+
+from tests.components.climate import common
+
+FAN_SETTINGS = [
+    FAN_OFF,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_HIGH,
+]
+
+HVAC_SETTINGS = [
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    HVAC_MODE_AUTO,
+]
+
+ENTITY_CLIMATE = "climate.fakespa_climate"
+
+
+async def test_spa_defaults(hass: HomeAssistant):
+    """Test supported features flags."""
+
+    await _setup_climate_test(hass)
+
+    state = hass.states.get(ENTITY_CLIMATE)
+
+    assert (
+        state.attributes["supported_features"]
+        == SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    )
+    assert state.state == HVAC_MODE_HEAT
+    assert state.attributes[ATTR_MIN_TEMP] == 10.0
+    assert state.attributes[ATTR_MAX_TEMP] == 40.0
+    assert state.attributes[ATTR_PRESET_MODE] == "Ready"
+    assert state.attributes[ATTR_HVAC_ACTION] == CURRENT_HVAC_IDLE
+
+
+async def test_spa_defaults_fake_tscale(hass: HomeAssistant):
+    """Test supported features flags."""
+
+    with patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.get_tempscale", return_value=1
+    ):
+        await _setup_climate_test(hass)
+
+    state = hass.states.get(ENTITY_CLIMATE)
+
+    assert (
+        state.attributes["supported_features"]
+        == SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    )
+    assert state.state == HVAC_MODE_HEAT
+    assert state.attributes[ATTR_MIN_TEMP] == 10.0
+    assert state.attributes[ATTR_MAX_TEMP] == 40.0
+    assert state.attributes[ATTR_PRESET_MODE] == "Ready"
+    assert state.attributes[ATTR_HVAC_ACTION] == CURRENT_HVAC_IDLE
+
+
+async def test_spa_with_blower(hass: HomeAssistant):
+    """Test supported features flags."""
+
+    with patch(
+        "homeassistant.components.balboa.BalboaSpaWifi.have_blower", return_value=True
+    ):
+        config_entry = await _setup_climate_test(hass)
+
+    # force a refresh
+    async_dispatcher_send(hass, SIGNAL_UPDATE.format(config_entry.entry_id))
```

Comment:
Please patch the library to get access to the callback instead, ie the callback can be gotten from the call_args of the patch mock. Then we can call the callback in the test to trigger a refresh. We don't want to access integration details in the tests.